File: admin/src/lib/types/seat/seat.dto.ts
export interface SeatDto {
  id: string;
  name: string;
  displayOrder: number;
  groupId: string;
  offset: SeatOffset;
}

export interface SeatOffset {
  x: number;
  y: number;
}

export interface SeatGroup {
  id: string;
  name: string;
  color: string;
  displayOrder: number;
  seats: SeatDto[];
}

export interface SeatGroupAddRequest {
  name: string;
  color: string;
}

export interface SeatGroupEditRequest extends SeatGroupAddRequest{
  id: string;
}

File: application/src/main/kotlin/dev/dornol/ticket/site/port/in/dto/SeatDto.kt
package dev.dornol.ticket.site.port.`in`.dto

import dev.dornol.ticket.site.domain.SeatGroupId
import dev.dornol.ticket.site.domain.SeatId

data class SeatDto(
    val id: SeatId,
    val name: String,
    val offset: SeatOffsetDto,
    val displayOrder: Long,
    val groupId: SeatGroupId
)
